modules/storages/spec/factories/webdav_data_factory.rb

```diff
@@ -28,9 +28,16 @@
 
 FactoryBot.define do
   factory :webdav_data, class: 'String' do
+    transient do
+      root_path { '' }
+      parent_path { '' }
+    end
+
     skip_create
 
     initialize_with do
+      base_path = File.join(root_path, '/remote.php/dav/files/admin', parent_path)
```

Comment:
Again, let's think about not using the `File` API here.

Reply:
for testing context it might even give you the flexibility to write a bit more comprehensive testing code.